src/main/java/uk/gov/justice/digital/delius/service/DeliusDocumentsService.java

```diff
@@ -0,0 +1,93 @@
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.justice.digital.delius.controller.BadRequestException;
+import uk.gov.justice.digital.delius.controller.NotFoundException;
+import uk.gov.justice.digital.delius.data.api.CommunityOrPrisonOffenderManager;
+import uk.gov.justice.digital.delius.data.api.UploadedDocumentCreateResponse;
+import uk.gov.justice.digital.delius.data.api.deliusapi.NewContact;
+import uk.gov.justice.digital.delius.data.api.deliusapi.UploadedDocumentDto;
+import uk.gov.justice.digital.delius.jpa.standard.repository.ContactTypeRepository;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+
+@Service
+public class DeliusDocumentsService {
+    private static final String EASU = "EASU";
+
+    private final DeliusApiClient deliusApiClient;
+    private final ContactTypeRepository contactTypeRepository;
+    private final OffenderManagerService offenderManagerService;
+
+    @Autowired
+    public DeliusDocumentsService(
+        DeliusApiClient deliusApiClient,
+        ContactTypeRepository contactTypeRepository,
+        OffenderManagerService offenderManagerService) {
+        this.deliusApiClient = deliusApiClient;
+        this.contactTypeRepository = contactTypeRepository;
+        this.offenderManagerService = offenderManagerService;
+    }
+
+    public UploadedDocumentCreateResponse createUPWDocument(String crn, Long convictionId, String contactTypeCode, MultipartFile document) {
+        assertCompletedUPWAssessmentContactType(contactTypeCode);
+
+        final var newContact= makeNewContact(crn, convictionId, contactTypeCode);
+        final var contactDto= deliusApiClient.createNewContact(newContact);
+
+        UploadedDocumentDto uploadedDocumentDto = deliusApiClient.uploadDocument(crn, contactDto.getId(), document);
+        return makeResponse(uploadedDocumentDto);
+    }
+
+    private UploadedDocumentCreateResponse makeResponse(UploadedDocumentDto uploadedDocumentDto) {
+        return new UploadedDocumentCreateResponse(
+            uploadedDocumentDto.getId(),
+            uploadedDocumentDto.getDocumentName(),
+            uploadedDocumentDto.getCrn(),
+            uploadedDocumentDto.getDateLastModified(),
+            uploadedDocumentDto.getLastModifiedUser(),
+            uploadedDocumentDto.getCreationDate()
+        );
+    }
+
+    private void assertCompletedUPWAssessmentContactType(String contactTypeCode) {
+        final var type = this.contactTypeRepository.findByCode(contactTypeCode)
+            .orElseThrow(() -> new BadRequestException(format("contact type '%s' does not exist", contactTypeCode)));
+
+        if (!type.getCode().equals(EASU)) {
+            throw new BadRequestException(format("contact type '%s' is not a completed UPW assessment type", contactTypeCode));
+        }
+    }
+
+    private NewContact makeNewContact(String crn, Long convictionId, String contactType) {
```

Comment:
It might be better to name this specifically as something like `makeNewUPWContact` and use the constant contact type value as it's just wrapping the builder